design/FY2017/NFP-Modeling LWR Heat Exchange between Buildings.md

```diff
@@ -0,0 +1,225 @@
+
+Note that $$F_{sky}+ F_{s1}+⋯+ F_{sn}+ F_{g}=1$$
+
+However, to avoid the complexity of iteratively calculating the long-wave radiation between building surfaces, we would simplify the case by using the pre-calculated surface temperature at the last time step for the current time step surface heat balance calculation. Considering the temperature of building exterior surfaces would not change too much between the last and current time steps, this simplification may sacrifice little accuracy but significantly improve the computing performance. To enable this, in Energy+.idd, we would add a variable Field: Calculate long-wave radiant heat exchange with surrounding surfaces in the object BuildingSurface:Detailed to indicate whether the heat exchange with surrounding surfaces is considered in the calculation. We would add an object SurfaceProperty:ViewFactors defining the view factors and another new object SurfaceProperty:SurroundingSurfaceTemperatures defining the outside surface temperature of these surrounding surfaces. View factors are assumed to be constant values, which can be overwritten in the EnergyPlus initialization stage by reading the calculation results of the external LWR calculation tool. The outside surface temperature can be exchanged with the external tool and be overwritten at each time step in EnergyPlus run time. In the building surface object, we use the Field: Surrounding Surfaces View Factor Name and Field: Surrounding Surfaces Temperature Name to link these objects. 
+
+	BuildingSurface:Detailed,
+	       \memo Allows for detailed entry of building heat transfer surfaces.  Does not include subsurfaces such as windows or doors.
+	       \extensible:3 -- duplicate last set of x,y,z coordinates (last 3 fields), remembering to remove ; from "inner" fields.
+	       \format vertices
+	       \min-fields 19
+	  A1 , \field Name
+	  …
+	  N1,  \field View Factor to Ground
+	       \type real
+	       \note From the exterior of the surface
+	       \note Unused if one uses the "reflections" options in Solar Distribution in Building input
+	       \note unless a DaylightingDevice:Shelf or DaylightingDevice:Tubular object has been specified.
+	       \note autocalculate will automatically calculate this value from the tilt of the surface
+	       \autocalculatable
+	       \minimum 0.0
+	       \maximum 1.0
+	       \default autocalculate
+	  A9,  \field Calculate long-wave radiant heat exchange with surrounding surfaces
+	       \type choice
+	       \key yes
+	       \key no
+	       \default no
+	       \note when yes, A10 and A11 are required. 
+	  A10, \field Surrounding Surfaces View Factor Name
+	       \type object-list
+	       \object-list SurroundingSurfacesViewFactorNames
+	       \note 
+	  A11, \field Surrounding Surfaces Temperature Name
+	       \type object-list
+	       \object-list SurroundingSurfaceTemperatures
+	       \note 
+	  N2 , \field Number of Vertices
+	  …
+	
+	SurfaceProperty:ViewFactors,
```

Comment:
> Is there anyway to input/override the ground or sky temperature?

I think the ground/sky temperatures are overridden in the next object.  Which I was going to ask about as well.  What if the user just wants to use the current ground surface and sky temperatures already calculated within E+.  The fields should be optional so they only need to be modified if the user actually wants to input a schedule.  

> Right now they are both position based

I also dislike the having to line up the view factors and schedules between two different objects in the correct positions.  Once a person is doing very detailed urban modeling and they have to line up (not that they'll be doing it manually) 20 different surrounding surfaces, it will be annoying.  I'd be more in favor of like:

```
 SurfaceProperty:SingleSurroundingSurface,
   A1, Name
   N1, ViewFactor
   A2, Temperature Schedule
```

Then in the building surface object, you have one collection object:

```
 SurfaceProperty:SurroundingSurfaces
   A1, Name
   N1, Sky View Factor
   N2, Sky Temp Override Schedule
   N3, Ground View Factor
   N4, Ground Temp Override Schedule
   A2, <name of a SurfaceProperty:SingleSurroundingSurface instance>
   A3, <name of a SurfaceProperty:SingleSurroundingSurface instance>
   A4, <name of a SurfaceProperty:SingleSurroundingSurface instance>
 - etc
```

Then in the BuildingSurface:Detailed object, you only have to add one new field:

```
Name of a SurfaceProperty:SurroundingSurfaces object instance
```

and if that is blank, you aren't doing any fancy surrounding surface temp stuff.  If it is not blank, you'll be able to walk the structure to get everything you need.

It will also set this code up to be enhanced later because we can easily add data to the `SurfaceProperty:SingleSurroundingSurface` object without having to create entirely new objects and arrays.